File: src/ios/TalkDocPlugin.m
#import "TalkDocPlugin.h"

#import <Cordova/CDVAvailability.h>

@implementation TalkDocPlugin

- (void)pluginInitialize {
}

- (void)saludar:(CDVInvokedUrlCommand*)command
{

    NSString* name = [[command arguments] objectAtIndex:0];
    NSString* msg = [NSString stringWithFormat: @"Hola todo el...  %@", name];

    CDVPluginResult* result = [CDVPluginResult
                               resultWithStatus:CDVCommandStatus_OK
                               messageAsString:msg];

    [self.commandDelegate sendPluginResult:result callbackId:command.callbackId];
}

- (void)llamar:(CDVInvokedUrlCommand*)command
{

    NSString* uuidUser = [[command arguments] objectAtIndex:0];
    NSString* idQbUser = [[command arguments] objectAtIndex:1];
    NSString* passQuickBloxUser = [[command arguments] objectAtIndex:2];
    NSString* email = [[command arguments] objectAtIndex:3];
    NSString* idQuickBloxDoctor = [[command arguments] objectAtIndex:4];
    NSString* msg = [NSString stringWithFormat: @"Hola estos son los params...  %@", uuidUser,
     @"IdQuickBloxUser %@", idQbUser, @"passQuickBloxUser %@", passQuickBloxUser,
     , @"email %@", email, @"idQuickBloxDoctor %@", idQuickBloxDoctor];

    CDVPluginResult* result = [CDVPluginResult
                               resultWithStatus:CDVCommandStatus_OK
                               messageAsString:msg];

    [self.commandDelegate sendPluginResult:result callbackId:command.callbackId];
}


@end

File: src/android/com/example/TalkDocPlugin.java
/**
 *
 */
package com.example;

import android.util.Log;
import org.apache.cordova.*;
import org.apache.cordova.PluginResult.Status;
import org.json.JSONArray;
import org.json.JSONException;

public class MiPlugin extends CordovaPlugin {
    private static final String TAG = "TalkDocPlugin";

    public void initialize(CordovaInterface cordova, CordovaWebView webView) {
        super.initialize(cordova, webView);

        Log.d(TAG, "Inicializando TalkDocPlugin");
    }

    public boolean execute(String action, JSONArray args, final CallbackContext callbackContext) throws JSONException {
        if (action.equals("saludar")) {
            String phrase = args.getString(0);
            final PluginResult result = new PluginResult(PluginResult.Status.OK, "Hola todo el... " + phrase);
            callbackContext.sendPluginResult(result);
        }

        if (action.equals("llamar")) {
            final String uuIdUser = args.getString(0);
            final String idQuickBloxUser = args.getString(1);
            final String passQuickBloxUser = args.getString(2);
            final String email = args.getString(3);
            final String idQuickBloxDoctor = args.getString(4);
            final PluginResult result = new PluginResult(PluginResult.Status.OK, "Hola estos son los params... " + "Uuid: " + uuIdUser +
                    "idQuick: " + idQuickBloxUser + " email: " + email + " idDoctor: " + idQuickBloxDoctor);
            callbackContext.sendPluginResult(result);
        };
        return true;
    }


}
